packages/docs-reanimated/docs/guides/worklets.mdx

```diff
@@ -0,0 +1,142 @@
+
+function App() {
+  const style = useAnimatedStyle(() => {
+    // Running on the UI thread
+    return { opacity: 0.5 };
+  });
+}
+```
+
+The `'worklet';` directive has no effect on the [JavaScript thread](/docs/fundamentals/glossary#javascript-thread). Worklets on the JavaScript thread are just functions.
+
+You can use [`runOnUI`](/docs/threading/runOnUI) to manually run worklets on the UI thread:
+
+```javascript
+function myWorklet() {
+  'worklet';
+  console.log('Hello from the UI thread');
+}
+
+function onPress() {
+  runOnUI(myWorklet)();
+}
+```
+
+You can pass data as arguments to worklets.
+
+```javascript
+function myWorklet(greeting) {
+  'worklet';
+  console.log(`${greeting} from the UI thread`);
+}
+
+function onPress() {
+  runOnUI(myWorklet)('Howdy');
+}
+```
+
+Worklets are [closures](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures). They can access variables declared outside of their own scope. Only variables referenced in the worklet body will be captured inside the worklet scope.
+
+```javascript
+const width = 135.5;
+
+function otherWorklet() {
+  'worklet';
+  console.log('Captured width is', width);
+}
+```
+
+Capturing large JavaScript objects inside of a worklet can lead to performance issues.
+
+```javascript
+const theme = {...}; // theme is a big object
+
+function myWorklet() {
+  'worklet';
+  console.log(theme.color); // 🚨 referenced `color` but captured the whole `theme` object
+}
+```
+
+You can get around this problem by first assigning the prop you want to use to a separate variable.
+
+```javascript
+const theme = {...};
+const color = theme.color;
+
+function myWorklet() {
+  'worklet';
+  console.log(color); // ✅ captured only `color`
+}
+```
+
+Worklets can return data within the same thread.
+
+```javascript
+function returningWorklet() {
+  'worklet';
+  return "I'm back"; // on the UI thread
+}
+
+function someWorklet() {
+  'worklet';
+  const what = returningWorklet(); // still on the UI thread
+  console.log('On the UI thread, other worklet says', what);
+}
+```
+
+To pass data between UI and JS thread we use [shared values](/docs/fundamentals/glossary#shared-value).
+
+```javascript
+import { useSharedValue } from 'react-native-reanimated';
+
+function App() {
+  const width = useSharedValue(100);
+
+  function myWorklet() {
+    'worklet';
+    width.value += 50;
+  }
+
+  width.value; // available on both JS and UI thread
+}
+```
+
+You can run functions on the JS thread from the UI thread with [`runOnJS`](/docs/threading/runOnJS). Most frequently used to call functions that aren't marked with a `'worklet';` directive (i.e. most third-party libraries) or to update the React state.
+
+```javascript
+import { router } from 'expo-router';
+import { Gesture } from 'react-native-gesture-handler';
+
+function App() {
+  const tap = Gesture.Tap().onEnd(() => {
+    // i'm a worklet too!
+    // highlight-next-line
+    runOnJS(router.back)();
+  });
+}
+```
```

Comment:
I'd expand this to annotate, that the function put in `runOnJS` has to be defined on the JS thread, i.e.:

```js
function worklet(){
  'worklet';
  runOnJS(() => console.log("Back to JS"))();
```

Will result in an error since the lambda function was created on the UI thread.

Reply:
Added in 3d50d09